add details on blob storage encryption (#14)
hashicorp/terraform-docs-common: website/docs/cloud-docs/architectural-details/data-security.mdx

## Vault Transit Encryption

The [Vault Transit Secret Engine](https://www.vaultproject.io/docs/secrets/transit/index.html)
handles encryption for data in-transit and is used when encrypting data from the
application to persistent storage.

## Blob Storage Encryption

> All objects persisted to blob storage are symmetrically encrypted prior to being
> written. Each object is encrypted with a unique encryption key. Objects are
> encrypted using 128 bit AES in CTR mode. The key material itself is round-tripped
> through the [Vault transit secret engine](https://www.vaultproject.io/docs/secrets/transit/index.html),
> which uses the default transit encryption cipher (AES-GCM with a 256-bit AES key
> and a 96-bit nonce), and stored alongside the object, in a pattern known as
> _envelope encryption_.

Reviewer: ```suggestion
All objects persisted to blob storage are symmetrically encrypted prior to being
written. Each object is encrypted with a unique encryption key. Objects are
encrypted using 128 bit AES in CTR mode. The key material is processed
through the [Vault transit secret engine](https://www.vaultproject.io/docs/secrets/transit/index.html),
which uses the default transit encryption cipher (AES-GCM with a 256-bit AES key
and a 96-bit nonce), and stored alongside the object. This pattern is called envelope encryption.
```

> Encryption key material is generated via the Vault transit secret engine's
> [datakey generation](https://www.vaultproject.io/api-docs/secret/transit#generate-data-key),
> using bit material from the kernel's cryptographically secure pseudo-random
> number generator (CSPRNG) as the `context` value. Because blob storage encryption
> generates a unique key for each object, and relies on envelope encryption,
> the encryption key material itself for any given object is not rotated.

Reviewer: ```suggestion
The Vault transit secret engine's
[datakey generation](https://www.vaultproject.io/api-docs/secret/transit#generate-data-key) creates the encryption key material using bit material from the kernel's cryptographically secure pseudo-random
number generator (CSPRNG) as the `context` value. Blob storage encryption generates a unique key for each object and relies on envelope encryption, so Vault does not rotate the encryption key material for individual objects.
```
